spicedb-as-library/main.go

```diff
@@ -0,0 +1,128 @@
+	defer cancel()
+
+	srv, err := newServer(ctx)
+	if err != nil {
+		log.Fatal("unable to init server: %w", err)
+	}
+
+	conn, err := connFromServer(ctx, srv)
+	if err != nil {
+		log.Fatal("unable to get gRPC connection: %w", err)
+	}
+
+	schemaSrv := v1.NewSchemaServiceClient(conn)
+	permSrv := v1.NewPermissionsServiceClient(conn)
+
+	go func() {
+		if err := srv.Run(ctx); err != nil {
+			log.Print("error while shutting down server: %w", err)
+		}
+	}()
+
+	_, err = schemaSrv.WriteSchema(ctx, &v1.WriteSchemaRequest{
+		Schema: `definition user {}
+                 definition resource {
+                   relation viewer: user
+                   permission view = viewer
+                 }`,
+	})
+	if err != nil {
+		log.Fatal("unable to get gRPC connection: %w", err)
+	}
+
+	resp, err := permSrv.WriteRelationships(ctx, &v1.WriteRelationshipsRequest{Updates: []*v1.RelationshipUpdate{
+		{
+			Operation: v1.RelationshipUpdate_OPERATION_CREATE,
+			Relationship: &v1.Relationship{
+				Resource: &v1.ObjectReference{
+					ObjectId:   "my_book",
+					ObjectType: "resource",
+				},
+				Relation: "viewer",
+				Subject: &v1.SubjectReference{
+					Object: &v1.ObjectReference{
+						ObjectId:   "john_doe",
+						ObjectType: "user",
+					},
+				},
+			},
+		},
+	}})
+	if err != nil {
+		log.Fatal("unable to get gRPC connection: %w", err)
+	}
+
+	token := resp.GetWrittenAt()
+	checkResp, err := permSrv.CheckPermission(ctx, &v1.CheckPermissionRequest{
+		Permission:  "view",
+		Consistency: &v1.Consistency{Requirement: &v1.Consistency_AtLeastAsFresh{AtLeastAsFresh: token}},
+		Resource: &v1.ObjectReference{
+			ObjectId:   "my_book",
+			ObjectType: "resource",
+		},
+		Subject: &v1.SubjectReference{
+			Object: &v1.ObjectReference{
+				ObjectId:   "john_doe",
+				ObjectType: "user",
+			},
+		},
+	})
+	if err != nil {
+		log.Fatal("unable to issue PermissionCheck: %w", err)
+	}
+
+	log.Printf("check result: %s", checkResp.Permissionship.String())
+}
+
+func connFromServer(ctx context.Context, srv server.RunnableServer) (*grpc.ClientConn, error) {
+	opts := []grpc.DialOption{
+		grpcutil.WithInsecureBearerToken("foobar"),
+	}
+
+	return srv.GRPCDialContext(ctx, opts...)
+}
+
+func newServer(ctx context.Context) (server.RunnableServer, error) {
+	ds, err := datastore.NewDatastore(ctx,
+		datastore.DefaultDatastoreConfig().ToOption(),
+		datastore.WithRequestHedgingEnabled(false),
+	)
+	if err != nil {
+		log.Fatalf("unable to start memdb datastore: %s", err)
+	}
+
+	configOpts := []server.ConfigOption{
+		server.WithGRPCServer(util.GRPCServerConfig{
+			Network: util.BufferedNetwork,
+			Enabled: true,
+		}),
+		server.WithPresharedSecureKey("foobar"),
+		server.WithHTTPGateway(util.HTTPServerConfig{HTTPEnabled: true}),
+		server.WithDashboardAPI(util.HTTPServerConfig{HTTPEnabled: false}),
+		server.WithMetricsAPI(util.HTTPServerConfig{HTTPEnabled: true}),
+		server.WithSilentlyDisableTelemetry(true),
```

Comment:
don't we still want telemetry? I mean it's fine if someone wants to turn it off but by default it's useful to have it.

Reply:
yeah that was an oversight